src/Controls/src/Core/Layout/Layout.cs

```diff
@@ -9,6 +9,144 @@
 
 namespace Microsoft.Maui.Controls
 {
+
+#pragma warning disable RS0016 // Add public types and members to the declared API
+	public abstract class Layout<T> : Layout where T : View, ICrossPlatformLayout
+	{
+		public new IList<T> Children { get; }
+		public Layout()
+		{
+			Children = new CastingList<T, IView>(this);
```

Comment:
Maybe this should rather be a field that directly wraps the base `_children` and just make that `private protected`? Then we have a field on this type that is wrapping the base type, and then return this new filed in `Children` instead of making the `Children` property actually store anything.

Reply:
I did a thing
I think I did what you were suggesting :-)